packages/react/src/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -1,29 +1,255 @@
 import {clsx} from 'clsx'
 import type {To} from 'history'
-import React from 'react'
+import React, {useState, useRef, useCallback, useEffect, useMemo} from 'react'
 import type {SxProp} from '../sx'
 import type {ComponentProps} from '../utils/types'
 import classes from './Breadcrumbs.module.css'
 import type {ForwardRefComponent as PolymorphicForwardRefComponent} from '../utils/polymorphic'
 import {BoxWithFallback} from '../internal/components/BoxWithFallback'
+import {ActionMenu} from '../ActionMenu'
+import {ActionList} from '../ActionList'
+import {useResizeObserver} from '../hooks/useResizeObserver'
+import type {ResizeObserverEntry} from '../hooks/useResizeObserver'
 
 const SELECTED_CLASS = 'selected'
 
 export type BreadcrumbsProps = React.PropsWithChildren<
   {
     className?: string
+    overflow?: 'wrap' | 'menu'
+    hideRoot?: boolean
   } & SxProp
 >
 
 const BreadcrumbsList = ({children}: React.PropsWithChildren) => {
   return <ol className={classes.BreadcrumbsList}>{children}</ol>
 }
 
-function Breadcrumbs({className, children, sx: sxProp}: BreadcrumbsProps) {
-  const wrappedChildren = React.Children.map(children, child => <li className={classes.ItemWrapper}>{child}</li>)
+type BreadcrumbsMenuItemProps = {
+  items: React.ReactElement[]
+  'aria-label'?: string
+}
+
+const BreadcrumbsMenuItem = React.forwardRef<HTMLButtonElement, BreadcrumbsMenuItemProps>(
+  ({items, 'aria-label': ariaLabel, ...rest}, ref) => {
+    return (
+      <ActionMenu>
+        <ActionMenu.Button
```

Comment:
@pksjce could you update this to use the `IconButton` with the `kebab` icon?

Reply:
I am struggling with the vertical align of this button now.
<img width="351" height="41" alt="Screenshot 2025-08-15 at 7 31 52 pm" src="https://github.com/user-attachments/assets/7bc6c90f-4798-4cd2-bde7-67464db37f65" />

Reply:
Done!